cpp/python/CMakeLists.txt

```diff
@@ -3,13 +3,78 @@
 # SPDX-License-Identifier: BSD 3-Clause License
 # ======================================================================================================
 
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+include(../../fetch_rapids.cmake)
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-export)
+include(rapids-find)
+
+file(READ "${CMAKE_CURRENT_LIST_DIR}/../../VERSION" _version_contents)
+if(_version_contents MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+).*$")
+  set(libucxx_version "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}")
+else()
+  string(REPLACE "\n" "\n  " _version_contents_formatted "${_version_contents}")
+  message(FATAL_ERROR "Could not determine ucxx version. Contents of VERSION file:\n  ${_version_contents_formatted}")
+endif()
+
+project(
+  UCXX_PYTHON
+  VERSION ${libucxx_version}
+  LANGUAGES C CXX
+)
+
+option(BUILD_SHARED_LIBS "Build UCXX Python shared libraries" ON)
+option(FIND_UCXX_CPP "Search for existing UCXX C++ installations before defaulting to local files"
+       OFF
+)
```

Comment:
I'm guessing that when `BUILD_SHARED_LIBS=ON` then `libucxx.so` is found transitively, but if `BUILD_SHARED_LIBS=OFF` then `FIND_UCXX_CPP=ON` is necessary, is that right?

Reply:
No. `BUILD_SHARED_LIBS` merely determines whether static (`.a`) or shared (`.so`) libraries are built. It has nothing to do with whether or not `FIND_UCXX_CPP` is set.

Reply:
Got it, but when is `FIND_UCXX_CPP` required here then? I assume it always needs to be found, unless maybe `libucxx_python.so` is doing that transitively, is that the case?

Reply:
`FIND_UCXX_CPP` is set to `ON` for the Conda package, which depends on the `libucxx` Conda package. It's currently set to `OFF` for the wheel, which instead builds `libucxx` through `add_subdirectory()`, but a future PR that creates a `libucxx` wheel will also set it to `ON`.

Reply:
I see now, thanks for clarifying.